benchmarks/nsinker_spherical_shell/nsinker.cc

```diff
@@ -0,0 +1,338 @@
+      centers[43] = Point<3>(4.3309508843e-02, 9.0291512474e-01, 2.9450144167e-01);
+      centers[44] = Point<3>(6.8573011443e-01, 6.6033273035e-02, 8.2121989495e-01);
+      centers[45] = Point<3>(2.4277445452e-01, 3.1025718772e-01, 4.9255406554e-01);
+      centers[46] = Point<3>(3.5617944848e-01, 3.0799053857e-01, 3.9698166931e-01);
+      centers[47] = Point<3>(7.0916077621e-02, 8.8651657239e-01, 6.8403214295e-01);
+      centers[48] = Point<3>(5.2822650202e-01, 9.0281945043e-01, 6.8650344000e-01);
+      centers[49] = Point<3>(6.3316007640e-02, 1.5214040370e-01, 2.3765034985e-02);
+      centers[50] = Point<3>(4.1894298765e-01, 1.7479340461e-01, 7.5275125343e-01);
+      centers[51] = Point<3>(4.9031640053e-01, 7.4774375406e-01, 3.2927456281e-01);
+      centers[52] = Point<3>(1.1757708859e-01, 1.1812786251e-01, 3.7498524244e-01);
+      centers[53] = Point<3>(3.7696964032e-01, 7.2874483733e-01, 1.4480990830e-02);
+      centers[54] = Point<3>(3.8201288152e-01, 4.9049964756e-01, 8.2757658503e-01);
+      centers[55] = Point<3>(7.9664661586e-02, 9.2396727806e-01, 1.1804237828e-01);
+      centers[56] = Point<3>(9.3825167927e-01, 1.9597347043e-01, 7.2611756191e-01);
+      centers[57] = Point<3>(8.5786301170e-01, 1.0363770514e-01, 8.3891028205e-01);
+      centers[58] = Point<3>(5.6511039453e-01, 8.1040084307e-01, 4.0696941614e-01);
+      centers[59] = Point<3>(9.3497714490e-01, 1.6087440083e-01, 8.1605472361e-01);
+      centers[60] = Point<3>(4.3173963829e-01, 2.4810082244e-01, 8.3052277138e-01);
+      centers[61] = Point<3>(5.9621858625e-01, 6.4577903070e-01, 6.0816894547e-01);
+      centers[62] = Point<3>(4.9546643556e-01, 3.0438243752e-01, 7.5562733447e-01);
+      centers[63] = Point<3>(8.2861043319e-01, 4.5555055302e-01, 4.3814466774e-01);
+      centers[64] = Point<3>(8.9743076959e-01, 1.1894442752e-01, 9.8993320995e-02);
+      centers[65] = Point<3>(6.9884936497e-01, 5.6127713367e-01, 3.8478565932e-01);
+      centers[66] = Point<3>(9.2576270966e-02, 9.2938612771e-01, 1.9264837596e-01);
+      centers[67] = Point<3>(8.4125479722e-01, 9.6937695284e-01, 3.1844636161e-01);
+      centers[68] = Point<3>(1.2799954700e-01, 2.8838638276e-01, 9.0993508972e-01);
+      centers[69] = Point<3>(2.7905288352e-01, 4.1813262758e-02, 7.5550716964e-01);
+      centers[70] = Point<3>(8.0900019305e-01, 8.6624463269e-01, 9.7354159503e-01);
+      centers[71] = Point<3>(3.1358765965e-01, 4.6779574243e-01, 2.4304298462e-01);
+      centers[72] = Point<3>(8.2344259034e-01, 5.9961585635e-01, 7.4369772512e-01);
+      centers[73] = Point<3>(3.2766604253e-01, 8.3176720460e-02, 9.5114077951e-01);
+      centers[74] = Point<3>(8.2308128282e-01, 5.2712029523e-01, 3.1080186614e-01);
+
+      // First rescale their location from [0,1] to [-1,1]:
+      for (auto &p: centers)
+        for (int d=0; d<3; ++d)
+          p[d] = p[d]*2.0-1.0;
+
+      // Now erase all points with center outside our shell:
```

Comment:
How do you make sure you don't have any sinkers that are only halfway in the model domain? I assume the initial locations were picked that way? Or is this not important?

Reply:
This means we keep some sinkers that are at least halfway inside the domain, yes. I added a comment.